tedana/io.py

```diff
@@ -16,6 +17,50 @@
 LGR = logging.getLogger(__name__)
 
 
+def gen_fname(basefile, extension='_bold.nii.gz', **kwargs):
+    """
+    Generate BIDS derivatives-compatible filename from components.
+
+    Parameters
+    ----------
+    basefile : :obj:`str`
+        Name of file from which to derive BIDSRawBase prefix and datatype
+        suffix.
+    extension : :obj:`str`, optional
+        Extension for file. Includes the datatype suffix. Default is
+        "_bold.nii.gz".
+    kwargs : :obj:`str`
+        Additional keyword arguments are added to the filename in the order
+        they appear.
+
+    Returns
+    -------
+    out_file : :obj:`str`
+        BIDS derivatives-compatible filename
+    """
+    if not all([isinstance(v, str) for k, v in kwargs.items()]):
+        raise ValueError("All keyword arguments must be strings")
+
+    # Remove echo field from filename
```

Comment:
Will this break if given a file without an 'echo' key ? For example, in `fmriprep` we currently remove the echo key after optimal combination, but prior to denoising.

Reply:
It seems to run fine when there's no echo key. My test:
```python
>>> import re
>>> echo_regex = re.compile('_echo-[0-9+]_')
>>> basefile1 = 'sub-01_ses-01_echo-1_run-01_bold.nii.gz'
>>> test1 = re.sub(echo_regex, '_', basefile1)
>>> test1
'sub-01_ses-01_run-01_bold.nii.gz'
>>> basefile2 = 'sub-01_ses-01_run-01_bold.nii.gz'
>>> test2 = re.sub(echo_regex, '_', basefile2)
>>> test2
'sub-01_ses-01_run-01_bold.nii.gz'
```